src/som/interpreter/nodes/nary/RedirectWriteNode.java

```diff
@@ -0,0 +1,49 @@
+package som.interpreter.nodes.nary;
+import com.oracle.truffle.api.frame.FrameSlot;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.compiler.Variable;
+import som.interpreter.nodes.ExpressionNode;
+import som.interpreter.nodes.PreevaluatedExpression;
+
+public class RedirectWriteNode extends ExprWithTagsNode
+    implements PreevaluatedExpression {
+
+  final VirtualFrame writeFrame;
+  final FrameSlot slot;
+  final ExpressionNode[] args;
+
+  public RedirectWriteNode(final VirtualFrame frame, final Variable var,
+      final ExpressionNode[] args, final SourceSection source) {
+    super(source);
+    this.writeFrame = frame;
+    this.args = args;
+
+    FrameSlot slotForVar = null;
+    for (FrameSlot slot : writeFrame.getFrameDescriptor().getSlots()) {
```

Comment:
the `Variable` class doesn't support writes, indeed. Not all variables allow assignment.
Only local variables are writable. Most importantly for you, Arguments are not assignable.

Here the relevant code in `Local`:
https://github.com/richard-roberts/SOMns/blob/implement/objectLiterals/src/som/compiler/Variable.java#L185